### PR TITLE
General Cleanup 2

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Just some General cleanup around the repo that I noticed.

Biggest things to note:
    * Added OPEN_DISPS EnBigpo_DrawLantern
    * Changed BgIkanaRay to actually use an ActionFunc
    * Better matches for func_80B9765C (EnZot) and ObjHsStump_Appear
    * Remove uneeded casts for both Graph_Alloc and Lib_SegmentedToVirtual
    * Booleans for BgCheck test functions

### CODE REVIEW

File: src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c [hunk 1]
```diff
@@ -1359,10 +1359,6 @@ void EnBigpo_DrawScoopSoul(Actor* thisx, GlobalContext* globalCtx) {
     CLOSE_DISPS(globalCtx->state.gfxCtx);
 }
 
-/*
- * this matches without OPENDISPS but with it has stack issues,
- *  might be able to find an alternative match with the macros, so far no success
- */
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c:1365]: So... what changed? I don't see anything special in the diff

File: src/overlays/actors/ovl_En_Dno/z_en_dno.c [hunk 2]
```diff
@@ -40,8 +40,6 @@ void func_80A732C8(EnDno* this, GlobalContext* globalCtx);
 s32 EnDno_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx);
 void EnDno_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx);
 
-extern u8 D_801C20C0;
-
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Dno/z_en_dno.c:44]: I see this symbol is defined in undefined_syms.txt too, don't forget to remove it from that file too

**Review by AngheloAlf — APPROVED** [review]
Nice cleanup

File: src/overlays/actors/ovl_En_Toto/z_en_toto.c [hunk 3]
```diff
@@ -250,7 +251,9 @@ void func_80BA39C8(EnToto* this, GlobalContext* globalCtx) {
         return;
     }
 
-    if ((globalCtx->sceneNum == SCENE_MILK_BAR && !(gSaveContext.time >= 0x4000 && gSaveContext.time < 0xED02)) ||
+    //! @TODO: 0xED02 nor 0xED01 match CLOCK_TIME macro
+    if ((globalCtx->sceneNum == SCENE_MILK_BAR &&
+         !(gSaveContext.time >= CLOCK_TIME(6, 0) && gSaveContext.time < 0xED02)) ||
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Toto/z_en_toto.c:256]: I assume the second one can't be written nicely?

File: src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c (hunk 1 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c:1365]: To be honest I have no idea. I was able to just use Macros with no issues.

File: src/overlays/actors/ovl_Obj_HsStump/z_obj_hsstump.c [hunk 4]
```diff
@@ -87,43 +87,43 @@ void ObjHsStump_Appear(ObjHsStump* this, GlobalContext* globalCtx) {
...
             s32 i;
-            f32 angle;
+            f32 angleDeg;
             s16 numDirections = 4;
             Vec3f iceSmokePosOffset;
             Vec3f iceSmokeVelOffset;
             s16 offsetAngle;
             Vec3f iceSmokeVel;
-            f32 angleBinary;
+            f32 angleBrad;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_HsStump/z_obj_hsstump.c:99]: I still don't like this as a name... if only because it's a name lol.

**Review by EllipticEllipsis — COMMENTED** [review]
Always a fan of less fake matches (and fewer fake matches)

File: src/overlays/actors/ovl_Bg_Ikana_Ray/z_bg_ikana_ray.c [hunk 5]
```diff
@@ -69,7 +69,7 @@ void BgIkanaRay_Init(Actor* thisx, GlobalContext* globalCtx) {
     Collider_SetCylinder(globalCtx, collision, &this->actor, &sCylinderInit);
     Collider_UpdateCylinder(&this->actor, &this->collision);
 
-    this->animatedTextures = (AnimatedMaterial*)Lib_SegmentedToVirtual(object_ikana_obj_Matanimheader_001228);
+    this->animatedTextures = Lib_SegmentedToVirtual(object_ikana_obj_Matanimheader_001228);
 
     if (Flags_GetSwitch(globalCtx, this->actor.params & 0x7F)) {
```
Comment (jpburnett) [inline src/overlays/actors/ovl_Bg_Ikana_Ray/z_bg_ikana_ray.c:74]: Since this file is already being changed, would it be convenient to make a macro for this? I understand these clean up PRs can have a lot of scope creep.

File: tools/disasm/functions.txt [hunk 6]
```diff
@@ -12208,7 +12208,7 @@
     0x80AD5394:("func_80AD5394",),
     0x80AD54C8:("EnTrt2_OverrideLimbDraw",),
     0x80AD5584:("EnTrt2_PostLimbDraw",),
-    0x80AD566C:("EnTrt2_UnkDraw",),
+    0x80AD566C:("EnTrt2_TransformLimbDraw",),
```
[the remark this answers is not in the document]
Comment (hensldm) [reply tools/disasm/functions.txt:12211]: It wasn't, thanks for the reminder.

File: src/overlays/actors/ovl_En_Go/z_en_go.c [hunk 7]
```diff
@@ -1449,7 +1449,11 @@ void func_80A146CC(EnGo* this, GlobalContext* globalCtx) {
 }
 
 void func_80A14798(EnGo* this, GlobalContext* globalCtx) {
-    s32 sp38[2] = { 0x0000003E, 0x00000F64 };
+    EffectTireMarkInit sp38 = {
```
Comment (jpburnett) [inline src/overlays/actors/ovl_En_Go/z_en_go.c:1452]: While I understand what this means, gorons aren't really tires... maybe skid marks as a name?

File: src/overlays/actors/ovl_En_Toto/z_en_toto.c (hunk 3 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Toto/z_en_toto.c:256]: No, that is what the TODO is for. The thing 0xED02 isn't even close to either boundary.
CLOCK_TIME(22, 13) = 0xECFA  and CLOCK_TIME(22, 14) = 0xED27

**Review by jpburnett — COMMENTED** [review]
Clean code is happy code, appreciate the work done here. Just a couple of thoughts

File: src/overlays/actors/ovl_Bg_Ikana_Ray/z_bg_ikana_ray.c (hunk 5 above)
Comment (hensldm) [reply src/overlays/actors/ovl_Bg_Ikana_Ray/z_bg_ikana_ray.c:74]: Thanks for pointing it out. Added the macro.

File: src/overlays/actors/ovl_En_Go/z_en_go.c (hunk 7 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Go/z_en_go.c:1452]: Don't really want to name things in my cleanup PRs, would rather wait for the whole file to be documented at once.

Comment (jpburnett) [reply src/overlays/actors/ovl_En_Go/z_en_go.c:1452]: That's fair 👍

**Review by jpburnett — APPROVED** [review]

File: src/overlays/actors/ovl_En_Toto/z_en_toto.c (hunk 3 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Toto/z_en_toto.c:256]: Wow, that's annoying. Wonder if it's more sensible with the mult and div the other way or something.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
